sqlx-mysql/src/connection/executor.rs

```diff
@@ -183,7 +183,11 @@ impl MySqlConnection {
             loop {
                 // query response is a meta-packet which may be one of:
                 //  Ok, Err, ResultSet, or (unhandled) LocalInfileRequest
-                let mut packet = self.inner.stream.recv_packet().await?;
+                let mut packet = self.inner.stream.recv_packet().await.inspect_err(|_| {
```

Comment:
Yeah, if you could find the exact error code that's returned, that would probably be better. This is a huge pessimization.

Keep in mind we also get errors here for "normal" errors like constraint violations.

This would leak any valid prepared statements on the server side.